dfsdfsd

### **QUESTION 1: What are the oldest and newest founding years?**

In [10]:
SELECT MIN(b1.year_founded) AS oldest, MAX(b2.year_founded) AS newest
FROM businesses AS b1
    JOIN businesses AS b2 -- Applying here a sel-join to create values from the same column.
    ON b1.business = b2.business
    -- SQL Server doesn't seem to support a shorter USING clause.

(1 row affected)

Total execution time: 00:00:00.041

oldest newest 578 1999

Wow! That's a lot of variation between countries. In one country, the oldest business was only founded in 1999. By contrast, the oldest business in the world was founded back in 578. That's pretty incredible that a business has survived for more than a millennium.

### **QUESTION 2: How many and which businesses were founded before 1000?**

In [11]:
SELECT COUNT(business)
FROM businesses
WHERE year_founded < 1000

(1 row affected)

Total execution time: 00:00:00.030

(No column name) 6

fdsdf

In [14]:
SELECT *
FROM businesses
WHERE year_founded < 1000
ORDER BY year_founded

(6 rows affected)

Total execution time: 00:00:00.015

business year_founded category_code country_code Kongo Gumi 578 CAT6 JPN St. Peter Stifts Kulinarium 803 CAT4 AUT Staffelter Hof Winery 862 CAT9 DEU Monnaie de Paris  864 CAT12 FRA The Royal Mint 886 CAT12 GBR Sean's Bar 900 CAT4 IRL

In [18]:
SELECT b.business, b.year_founded, c2.country, c1.category
FROM businesses AS b
    JOIN categories AS c1
    ON b.category_code = c1.category_code
         JOIN countries AS c2
         ON b.country_code = c2.country_code
WHERE b.year_founded < 1000
ORDER BY b.year_founded;

(6 rows affected)

Total execution time: 00:00:00.016

business year_founded country category Kongo Gumi 578 Japan Construction St. Peter Stifts Kulinarium 803 Austria Cafés, Restaurants & Bars Staffelter Hof Winery 862 Germany Distillers, Vintners, & Breweries Monnaie de Paris  864 France Manufacturing & Production The Royal Mint 886 United Kingdom Manufacturing & Production Sean's Bar 900 Ireland Cafés, Restaurants & Bars

### **QUESTION 3: What are the long standing categories?**

In [22]:
SELECT TOP 5 category, COUNT(b.business) AS n -- LIMIT clause also not supported by SQL Server.
FROM categories AS c
    JOIN businesses AS b
    ON c.category_code = b.category_code
GROUP BY category
ORDER BY n DESC

(5 rows affected)

Total execution time: 00:00:00.017

category n Banking & Finance 37 Distillers, Vintners, & Breweries 22 Aviation & Transport 19 Postal Service 16 Manufacturing & Production 15